IntelliTect.TestTools.WindowsTestWrapper/ExampleProjects/Notepad.AutomatedTests/Helpers/NotepadWindowHelpers.cs

```diff
@@ -10,6 +12,37 @@ public class NotepadWindowHelpers
         private readonly NotepadWindow _NotepadWindow = new NotepadWindow();
         private readonly SaveAsWindow _SaveAsWindow = new SaveAsWindow();
 
+        public bool ValidateTextFile(string location, string fileName, string expectedText)
+        {
+            bool comparisonPass = false;
+            string combinedPath = Path.Combine( location, fileName );
+            if ( combinedPath == expectedText )
+            {
+                comparisonPass = true;
+            }
+            return comparisonPass;
+        }
+
+        public void DeleteDocument(string location, string fileName)
+        {
+            var combinedPath = Path.Combine(location, fileName);
+            int retryAttempts = 0;
+            do
+            {
+                File.Delete(combinedPath);
+                if (!File.Exists(combinedPath))
+                {
+                    break;
+                }
+                Thread.Sleep(100);
+                retryAttempts++;
+            } while (retryAttempts < 10);
+            if (File.Exists(combinedPath))
```

Comment:
I mostly agree with @MarkMichaelis on the exceptions, however in this case I think it is by far preferable to this code. Because things outside of the program can make changes to the file system, the recommended approach is to attempt the IO action and check for any exceptions. The issue with the above code is if the file is still in the process of being saved, the code you have above, will throw an exception and you will never get to your retry logic (which is the exact case where you want to retry). 

From the [docs](https://msdn.microsoft.com/en-us/library/system.io.file.delete(v=vs.110).aspx) it would appears that there are two exception that could be thrown where you want to sleep and retry, IOException and UnauthorizedAccessException. The latter is the easy case because you can simply assume that if you seen that exception type you need to retry. The IOException is a little more tricky because to do it right you only want to retry in the cases where it is thrown because the other process still has a file handle open to it. For this we need to look a little closer at what [File.Delete](https://referencesource.microsoft.com/#mscorlib/system/io/file.cs,250) actually does. If you follow the code through you will end up [here](https://referencesource.microsoft.com/#mscorlib/system/io/__error.cs,135). Specifically notice that the error code that you care about is `Win32Native.ERROR_SHARING_VIOLATION`. This value is put into the HResult property of the IOException. So you also want to retry in the case where an IOException is thrown with HResult = 0x80070020 (look at the MakeHRFromErrorCode to see how I got this value). 
As a side note, this would be an excellent case to use the exception filters feature (keyword when in the catch) that was introduced in C# 6.0.